src/Symfony/Component/Ldap/Adapter/ExtLdap/Connection.php

```diff
@@ -85,6 +85,65 @@ public function bind(?string $dn = null, #[\SensitiveParameter] ?string $passwor
         $this->bound = true;
     }
 
+    /**
+     * @param string $password WARNING: When the LDAP server allows unauthenticated binds, a blank $password will always be valid
+     *
+     * @return void
+     */
+    public function sasl_bind(?string $dn = null, #[\SensitiveParameter] ?string $password = null, ?string $mech = null, ?string $realm = null, ?string $authc_id = null, ?string $authz_id = null, ?string $props = null): void
+    {
+        if (!function_exists('ldap_sasl_bind')) {
+            throw new LdapException('Library - missing SASL support');
+        }
+
+        if (!$this->connection) {
+            $this->connect();
+        }
+
+        if (false === @ldap_sasl_bind($this->connection, $dn, $password, $mech, $realm, $authc_id, $authz_id, $props)) {
+            $error = ldap_error($this->connection);
+            switch (ldap_errno($this->connection)) {
```

Comment:
let's use a match instead of a switch?

Reply:
The switch in sasl_bind() follows what is done in the bind() method. Do you want me to change both?

Reply:
that'd be nice yes